StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx

```diff
@@ -1208,7 +1209,22 @@ void StMuDstMaker::fillFst(StEvent* ev) {
     connectFstCollection();
     mStMuDst->set(this);
   }
-  mFstUtil->fillMuFst(mFstCollection,fstcol);
+
+  //raw hit data input
+  StFstCollection* fstCollectionPtr = 0;
+  if (IAttr("fstMuRawHit")){//True for storing FST Raw hits
+    TObjectSet* fstDataSet = (TObjectSet*)GetDataSet("fstRawHitAndCluster");
```

Comment:
The FST raw hits are taken directly from the raw hit maker, rather than StEvent.  This sets up a situation where the information in the MuDst is no longer a subset of the information in StEvent.

Please obtain the hits from StEvent.

Reply:
Done.